bot/exts/evergreen/wikipedia.py

```diff
@@ -42,43 +43,39 @@ class WikipediaSearch(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
-    async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List[str]]:
+    async def wiki_request(self, channel: TextChannel, search: str) -> List[str]:
         """Search wikipedia search string and return formatted first 10 pages found."""
         params = WIKI_PARAMS | {"srlimit": 10, "srsearch": search}
         async with self.bot.http_session.get(url=SEARCH_API, params=params) as resp:
-            if resp.status == 200:
-                raw_data = await resp.json()
-                number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
-
-                else:
-                    await channel.send(
-                        "Sorry, we could not find a wikipedia article using that search term."
+            if (status := resp.status) != 200:
+                log.info(f"Unexpected response `{status}` while searching wikipedia for `{search}`")
+                raise APIError("Wikipedia API", status)
+
+            raw_data = await resp.json()
+
+            if not raw_data.get("query"):
+                if error := raw_data.get("errors"):
+                    log.error(f"There was an error while communicating with the Wikipedia API: {error}")
+                raise APIError("Wikipedia API", status, error)
+
+            number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
+            lines = []
+            if number_of_results:
```

Comment:
`number_of_results` is not used anywhere else?
```suggestion
            lines = []
            if raw_data["query"]["searchinfo"]["totalhits"]:
```